daemon/daemon.go

```diff
@@ -64,16 +74,27 @@ func Start(d Daemon, wait bool) error {
 		// Wait for the process to exit
 		err = cmd.Wait()
 		return err
+	} else {
+		// If the process exits at some point later, do read it's
+		// exit status. This should not let it be a zombie.
+		done := make(chan error, 1)
+		go func() {
+			done <- cmd.Wait()
+			err := <-done
+			log.WithFields(log.Fields{
+				"name":   d.Name(),
+				"pid":    cmd.Process.Pid,
+				"status": err,
+			}).Debug("Daemon died.")
+		}()
 	}
 
-	// Check if pidfile exists
-	pid, err := ReadPidFromFile(d.PidFile())
-	if err == nil {
-		// Check if process is running
-		_, err := GetProcess(pid)
-		if err == nil {
-			return errors.ErrProcessAlreadyRunning
-		}
+	// Check if the daemon is running
+	// TODO: Need some form of waiting period or timeout here before
+	// returning success. The process may die shortly after spawn.
+	_, err = GetProcess(pid)
```

Comment:
`pid` would be uninitialized here. I think you should be using  `cmd.Process.Pid`